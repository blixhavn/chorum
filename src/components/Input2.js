import React from 'react';
import { TextInput, View } from 'react-native';
import { FontAwesomeIcon } from '@fortawesome/react-native-fontawesome';
import { colors } from '../api/constants';

const Input2 = (props) => {
  return (

    <View style={styles.loginField}>
      <FontAwesomeIcon style={styles.inputIcon} icon={props.icon} />
      <TextInput style={styles.input} {...props} />
    </View>
  );
};

const styles = {
  
  loginField: {
    flexDirection: 'row',
    borderRadius: 10,
    paddingLeft: 10,
    paddingRight: 10,
    marginLeft: 30,
    marginRight: 30,
    marginBottom: 10,
    marginTop: 10,
    backgroundColor: colors.white,
  },
  input: {
    flex: 1,
    fontSize: 20,
    padding: 15,
    lineHeight: 23
  },
  inputIcon: {
    color: 'grey',
    padding: 10,
    paddingRight: 20,
    alignSelf: 'center'
  }
};

export { Input2 };
