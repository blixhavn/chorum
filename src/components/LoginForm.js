import React, { Component } from 'react';
import { View, Text, Image, TouchableHighlight, StyleSheet, TextInput } from 'react-native';
import { connect } from 'react-redux';
import { Actions } from 'react-native-router-flux';
import { FontAwesomeIcon } from '@fortawesome/react-native-fontawesome';
import { faAt, faKey } from '@fortawesome/free-solid-svg-icons';
import { emailChanged, passwordChanged, loginUser } from '../actions';
import { Button, Spinner } from './common';
import { colors } from './common/constants';

export class LoginForm extends Component {

  render() {
    return (
      <View style={styles.body}>
        <Image
          source={require('../img/chorum-splash.png')}
          style={styles.image}
        />
        <Text style={styles.errorTextStyle}>
          {this.props.error}
        </Text>
        <View style={styles.loginField}>
          <FontAwesomeIcon style={styles.inputIcon} icon={faAt} />
          <TextInput
            style={styles.input}
            autoCapitalize='none'
            placeholder="your@email.com"
            onChangeText={this.onEmailChange.bind(this)}
            value={this.props.email}
            keyboardType="email-address"
            autoCompleteType="email"
            textContentType="emailAddress"
          />
        </View>

        <View style={styles.loginField}>
          <FontAwesomeIcon style={styles.inputIcon} icon={faKey} />
          <TextInput
            style={styles.input}
            secureTextEntry
            placeholder="**********"
            onChangeText={this.onPasswordChange.bind(this)}
            value={this.props.password}
            textContentType="password"
            autoCompleteType="password"
          />
        </View>

        {this.renderButton()}

        <TouchableHighlight  style={styles.bottomLink} onPress={() => { Actions.register() }}>
          <Text style={styles.registerText}>Register new account</Text>
        </TouchableHighlight>
      </View>
    );
  }

  onEmailChange(text) {
    this.props.emailChanged(text);
  }

  onPasswordChange(text) {
    this.props.passwordChanged(text);
  }

  onButtonPress() {
    const { email, password } = this.props;

    this.props.loginUser({ email, password });
  }

  onRegisterLinkPress() {
    Action
  }

  renderButton() {
    if (this.props.loading) {
      return <Spinner style={{ flex: null, padding: 23 }} size="large" color={colors.green} />;
    }

    return (
      <Button
        buttonStyle={styles.loginButton}
        textStyle={styles.loginButtonText}
        onPress={this.onButtonPress.bind(this)}
      >
        Login
      </Button>
    );
  }
}


const styles = StyleSheet.create({
  body: {
    flex: 1,
    flexDirection: 'column',
    backgroundColor: colors.purple,
    justifyContent: 'flex-start'
  },
  loginField: {
    flexDirection: 'row',
    borderRadius: 10,
    paddingLeft: 10,
    paddingRight: 10,
    marginLeft: 30,
    marginRight: 30,
    marginBottom: 10,
    marginTop: 10,
    backgroundColor: colors.white,
  },
  input: {
    flex: 1,
    fontSize: 20,
    padding: 15
  },
  inputIcon: {
    color: 'grey',
    padding: 10,
    paddingRight: 20,
    alignSelf: 'center'
  },
  loginButton: {
    flex: null,
    borderRadius: 10,
    backgroundColor: colors.green,
    borderWidth: 0,
    padding: 10,
    marginLeft: 30,
    marginRight: 30,
    marginTop: 10,
    marginBottom: 10,
  },
  loginButtonText: {
    color: colors.grey,
    fontSize: 18,
    fontWeight: '600'
  },
  errorTextStyle: {
    fontSize: 20,
    alignSelf: 'center',
    color: 'red'
  },
  bottomLink: {
    alignContent: 'center',
    paddingBottom: 15
  },
  registerText: {
    fontSize: 18,
    alignSelf: 'center',
    color: colors.white,
  },
  image: {
    width: 300,
    height: 300,
    alignSelf: 'center',
  },
  smallCard: {
    paddingLeft: 30,
    paddingRight: 30
  }
})


const mapStateToProps = ({ auth }) => {
  const { email, password, error, loading } = auth;

  return { email, password, error, loading };
};

export default connect(mapStateToProps, {
  emailChanged, passwordChanged, loginUser
})(LoginForm);
