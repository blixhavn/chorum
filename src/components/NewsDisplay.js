import React, { Component } from 'react';
import { Text } from 'react-native';
import { connect } from 'react-redux';
import { Card, CardSection } from './common';

export class NewsDisplay extends Component {

  render() {
    return (
      <Card>
        <CardSection>
          <Text>News goes here. Soon. Very soon. Sooooon</Text>
        </CardSection>
      </Card>
    );
  }
}


export default connect(null)(NewsDisplay);
