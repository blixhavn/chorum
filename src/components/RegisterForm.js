import React, { Component } from 'react';
import { Text } from 'react-native';
import { connect } from 'react-redux';
import { regEmailChanged, regPasswordChanged, registerUser } from '../actions';
import { Card, CardSection, Input, Button, Spinner } from './common';

class RegisterForm extends Component {
  onEmailChange(text) {
    this.props.regEmailChanged(text);
  }

  onPasswordChange(text) {
    this.props.regPasswordChanged(text);
  }

  onButtonPress() {
    const { email, password } = this.props;

    this.props.registerUser({ email, password });
  }

  renderButton() {
    if (this.props.loading) {
      return <Spinner size="large" />;
    }

    return (
      <Button onPress={this.onButtonPress.bind(this)}>
        Register
      </Button>
    );
  }

  render() {
    return (
      <Card>
        <CardSection>
          <Input
            label="Email"
            placeholder="email@gmail.com"
            onChangeText={this.onEmailChange.bind(this)}
            value={this.props.email}
          />
        </CardSection>

        <CardSection>
          <Input
            secureTextEntry
            label="Password"
            placeholder="password"
            onChangeText={this.onPasswordChange.bind(this)}
            value={this.props.password}
          />
        </CardSection>

        <Text style={styles.errorTextStyle}>
          {this.props.error}
        </Text>

        <CardSection>
          {this.renderButton()}
        </CardSection>
      </Card>
    );
  }
}

const styles = {
  errorTextStyle: {
    fontSize: 20,
    alignSelf: 'center',
    color: 'red'
  }
};

const mapStateToProps = ({ register }) => {
  const { email, password, error, loading } = register;

  return { email, password, error, loading };
};

export default connect(mapStateToProps, {
  regEmailChanged, regPasswordChanged, registerUser
})(RegisterForm);
