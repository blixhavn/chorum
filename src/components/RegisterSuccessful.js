import React, { Component } from 'react';
import { Text, TouchableHighlight } from 'react-native';
import { connect } from 'react-redux';
import { Card, CardSection } from './common';
import { Actions } from 'react-native-router-flux';

class RegisterSuccessful extends Component {

  render() {
    return (
      <Card>
      <CardSection>
        <Text>You're now registered.</Text>
      </CardSection>
      
      <CardSection>
          <TouchableHighlight onPress={() => Actions.reset('auth')}>
            <Text>You can now log in</Text>
          </TouchableHighlight>
        </CardSection>
        
      </Card>
    );
  }
}


export default connect(null)(RegisterSuccessful);
