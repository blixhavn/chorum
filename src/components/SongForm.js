import React, { Component } from 'react';
import { View, Text } from 'react-native';
import { connect } from 'react-redux';
import { Card, CardSection } from './common';

class SongForm extends Component {
    render() {
        return (
            <Card>
              <CardSection>
                <Text>Create song</Text>
              </CardSection>
            </Card>
        );
    }
}

export default connect(null)(SongForm);