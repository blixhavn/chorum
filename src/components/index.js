export * from './Button';
export * from './Button2';
export * from './Card';
export * from './CardSection';
export * from './Header';
export * from './Input';
export * from './Input2';
export * from './Spinner';
export * from './Confirm';
export * from './TabIcon';
