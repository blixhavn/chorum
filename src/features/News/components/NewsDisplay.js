import React, { Component } from 'react';
import { Text } from 'react-native';
import { connect } from 'react-redux';
import { Card, CardSection } from '../../../components';

export class NewsDisplay extends Component {

  render() {
    return (
      <Card>
        <CardSection>
          <Text>News goes here. Soon. Very soon. Sooooon</Text>
        </CardSection>
      </Card>
    );
  }
}

export const TabIcon = ({selected, title}) => {
    return (
        <Text style={{color: selected ? 'red' : 'black'}}>{title}</Text>
    )
}


export default connect(null)(NewsDisplay);
