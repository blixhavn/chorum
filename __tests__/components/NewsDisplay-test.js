import React from 'react';
import 'react-native';
import renderer from 'react-test-renderer';
import { NewsDisplay } from '../../src/components/NewsDisplay';

describe("NewsDisplay Component", () => {
    it("renders correctly", () => {
      const loginForm = renderer.create(<NewsDisplay />).toJSON();
      expect(loginForm).toMatchSnapshot();
    });
});
