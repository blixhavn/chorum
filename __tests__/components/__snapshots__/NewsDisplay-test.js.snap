// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsDisplay Component renders correctly 1`] = `
<View
  style={
    Object {
      "borderBottomWidth": 0,
      "borderColor": "#ddd",
      "borderRadius": 2,
      "borderWidth": 1,
      "elevation": 1,
      "marginLeft": 5,
      "marginRight": 5,
      "marginTop": 10,
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.1,
      "shadowRadius": 2,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
          "borderBottomWidth": 1,
          "borderColor": "#ddd",
          "flexDirection": "row",
          "justifyContent": "flex-start",
          "padding": 5,
          "position": "relative",
        },
        undefined,
      ]
    }
  >
    <Text>
      News goes here. Soon. Very soon. Sooooon
    </Text>
  </View>
</View>
`;
